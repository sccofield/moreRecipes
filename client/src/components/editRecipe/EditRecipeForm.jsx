import React from 'react';
import Dropzone from 'react-dropzone';


const PostRecipeForm = props => (
  <main className="container">

    <div className="row justify-content-md-center">
      <div className="col-md-8">

        <h1>Edit Recipe</h1>
        {props.state.errors && <ul className="errorMessage">{props.state.errors.map(error => <li key={Math.floor(Math.random() * 1000)}>{error}</li>)}</ul>}

        {
          props.state &&
          <form onSubmit={props.onSubmit}>
            <div className="form-group">
              <label>Recipe title</label>
              <input
                type="text"
                className="form-control"
                name="title"
                id="recipeTitle"
                placeholder="Enter recipe title"
                value={props.state.title}
                onChange={props.onChange}
                required
              />
            </div>
            <div className="form-group">
              <label htmlFor="post">Recipe ingredients</label>

              <textarea
                className="form-control"
                name="ingredients"
                id="ingredients"
                placeholder="Enter recipe ingredients seperated by commas"
                rows="5"
                value={props.state.ingredients}
                onChange={props.onChange}
                required
              />
              <small id="emailHelp" className="form-text text-muted">seperate each ingredients with a comma (,)</small>

            </div>

            <div className="form-group">
              <label htmlFor="post">Recipe description</label>
              <textarea
                className="form-control"
                name="description"
                id="description"
                rows="5"
                placeholder="Enter recipe description"
                value={props.state.description}
                onChange={props.onChange}
                required
              />
            </div>

            <div className="form-group">
              <label htmlFor="imgurl">Recipe image</label>
              <Dropzone
                onDrop={props.handleDrop}
                multiple={false}
              >
                <p className="text-center pt-4">
                Click to upload image
                </p>
                { props.state.imageUrl &&

                <div className="form-group">
                  <img src={props.state.imageUrl} alt="" className="img img-responsive img-thumbnail imagePreview" />
                </div>
                }
              </Dropzone>


            </div>


            <button type="submit" className="btn btn-primary">Submit</button>
          </form>
        }

      </div>
    </div>
  </main>

);

export default PostRecipeForm;
