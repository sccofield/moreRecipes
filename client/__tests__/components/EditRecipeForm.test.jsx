import React from 'react';
import toJSON from 'enzyme-to-json';
import EditRecipeForm from '../../src/components/editRecipe/EditRecipeForm';

describe('Recipe component', () => {
  const props = {
    state: {
      description: 'description',
      image: 'image',
      ingredients: 'ingredients',
      title: 'title'
    } 
  };

  const errorProps = {
    state: {
      description: 'description',
      image: 'image',
      ingredients: 'ingredients',
      title: 'title',
      errors: ['error']
    } 
  };

  const imageProps = {
    state: {
      description: 'description',
      imageUrl: 'image',
      ingredients: 'ingredients',
      title: 'title',
    },
    imagePreview: 'image'
  };

  
  it('should render editRecipe form when props is given', () => {
    const wrapper = shallow(<EditRecipeForm {...props} />);
    expect(wrapper.find('div').length).toEqual(6);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should show error message when there is an error in state', ()=> {
    const wrapper = shallow(<EditRecipeForm {...errorProps } />);
    expect(wrapper.find('.errorMessage').text()).toBe('error');
    expect(wrapper.find('.errorMessage li').length).toEqual(1);
    
  });

  it('should show the image if it has been uploaded', ()=> {
    const wrapper = shallow(<EditRecipeForm {...imageProps } />);
    expect(wrapper.find('.imagePreview').length).toEqual(1);
    
  });

});

