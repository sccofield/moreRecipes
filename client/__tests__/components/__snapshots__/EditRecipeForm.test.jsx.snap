// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe component should render editRecipe form when props is given 1`] = `
<main
  className="container"
>
  <div
    className="row justify-content-md-center"
  >
    <div
      className="col-md-8"
    >
      <h1>
        Edit Recipe
      </h1>
      <form>
        <div
          className="form-group"
        >
          <label>
            Recipe title
          </label>
          <input
            className="form-control"
            id="recipeTitle"
            name="title"
            placeholder="Enter recipe title"
            required={true}
            type="text"
            value="title"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="post"
          >
            Recipe ingredients
          </label>
          <textarea
            className="form-control"
            id="ingredients"
            name="ingredients"
            placeholder="Enter recipe ingredients seperated by commas"
            required={true}
            rows="5"
            value="ingredients"
          />
          <small
            className="form-text text-muted"
            id="emailHelp"
          >
            seperate each ingredients with a comma (,)
          </small>
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="post"
          >
            Recipe description
          </label>
          <textarea
            className="form-control"
            id="description"
            name="description"
            placeholder="Enter recipe description"
            required={true}
            rows="5"
            value="description"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="imgurl"
          >
            Recipe image
          </label>
          <t
            disableClick={false}
            disablePreview={false}
            disabled={false}
            maxSize={Infinity}
            minSize={0}
            multiple={false}
            preventDropOnDocument={true}
          >
            <p
              className="text-center pt-4"
            >
              Click to upload image
            </p>
          </t>
        </div>
        <button
          className="btn btn-primary"
          id="recipeSubmit"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</main>
`;
